### ISSUE TITLE
Empty Chat Messages Appearing Chat Thread

### ISSUE THREAD

**Issue by ramneettung-msft** [Azure/communication-ui-library#4320]
**Describe the bug; what happened?**

When using the latest beta version, empty chat messages are appearing in the chat thread, this differs from what is happening on the latest production version.

**What are the steps to reproduce the issue?**

Send empty chat message.

**What behavior did you expect?**

Empty chat message should not appear.

**If applicable, provide screenshots:**

**In what environment did you see the issue?**

    "@azure/communication-administration": "^1.0.0-beta.4",
    "@azure/communication-calling": "^1.22.2-beta.1",
    "@azure/communication-chat": "^1.5.0-beta.1",
    "@azure/communication-common": "^2.3.0",
    "@azure/communication-identity": "^1.3.0",
    "@azure/communication-react": "^1.14.0-beta.1",
    "@azure/communication-signaling": "^1.0.0-beta.23",

**Is there any additional information?**

Issue is that in the message metadata - fileSharingMetadata as default "[]" and messages always have attachments set as "[]" for attachments.

There is an if statement in the messageThreadSelector that should account for empty arrays in fileSharingMetadata and attachments fields:
 
**Potential Fix:**

```
if( (message.metadata?.fileSharingMetadata && message.metadata?.fileSharingMetadata.length !== '[]') ||
    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ (message.content?.attachments && message.content?.attachments.length !== 0)
  ) {
    return true;
  }
```

**Comment by emlynmac** [comment]
Hi @ramneettung-msft, thanks for raising the issue.
Can you include some steps to reproduce the situation so we can look into the issue?

**Comment by ramneettung-msft** [comment]
1. Using the chat composite
2. Send empty message:
```
    await adapter.sendMessage("");
```

3.

<img width="520" alt="Screenshot 2024-03-22 114948" src="https://github.com/Azure/communication-ui-library/assets/99354214/e77e10a2-3f81-4157-ab58-7eb96f7198f3">

**Comment by emlynmac** [comment]
@ramneettung-msft The Composite and SendBox Components won't let empty messages be sent.

Is there a use case that specifically needs an empty message to be sent through here?

**Comment by emlynmac** [comment]
@ramneettung-msft 
Please use the branch on pull #4322 to verify the fix for this.

**Comment by vhuseinova-msft** [comment]
Hi @ramneettung-msft, 
The fix is released in 1.15.0-beta.1. Please check it out and let us know if you have any questions.
